osu.Game/Rulesets/Ruleset.cs

```diff
@@ -46,6 +54,51 @@ public IEnumerable<Mod> GetAllMods() => Enum.GetValues(typeof(ModType)).Cast<Mod
                                                     // Resolve MultiMods as their .Mods property
                                                     .SelectMany(mod => (mod as MultiMod)?.Mods ?? new[] { mod });
 
+        private static readonly ConcurrentDictionary<int, Mod[]> mod_reference_cache = new ConcurrentDictionary<int, Mod[]>();
+
+        /// <summary>
+        /// Returns all mods for a query-only purpose.
+        /// Bindables should not be considered usable when retrieving via this method (use <see cref="GetAllMods"/> instead).
+        /// </summary>
+        public IEnumerable<Mod> GetAllModsForReference()
```

Comment:
How about `Unsafe` instead of `ForReference`?

Reply:
I'm fine with either. Will wait for a third opinion.

Reply:
Can agree with `Unsafe`. I'd ideally like as many alarm bells going on as possible when someone uses this, so that it's less likely that I forget to check if it's fine to use this version, and renaming achieves that for me.